src/tests/ftest/scripts/post_process_xml.sh

```diff
@@ -24,6 +24,13 @@ FILES=("$@")
 for file in "${FILES[@]}"; do
     if [ -f "$file" ]; then
         echo "Processing XML $file"
+        dir=$(dirname "$file")
+        base=$(basename "$file")
+        if [ "${base}" =~ "${COMP}" ]; then
```

Comment:
(lint) Can't use =~ in [ ]. Use [[..]] instead. [SC2074]
(lint) Remove quotes from right-hand side of =~ to match as a regex rather than literally. [SC2076]